Exiled.Events/Patches/Events/Scp096/TryingNotToCry.cs

```diff
@@ -25,28 +27,43 @@ namespace Exiled.Events.Patches.Events.Scp096
     [HarmonyPatch(typeof(Scp096), nameof(Scp096.TryNotToCry))]
     internal static class TryingNotToCry
     {
-        private static readonly int DoorMask = LayerMask.GetMask("Door");
-
-        private static bool Prefix(Scp096 __instance)
+        private static IEnumerable<CodeInstruction> Transpiler(IEnumerable<CodeInstruction> instructions, ILGenerator generator)
         {
-            if (!Physics.Raycast(__instance.Hub.PlayerCameraReference.position, __instance.Hub.PlayerCameraReference.forward, out RaycastHit hitInfo, 1f, DoorMask))
-                return false;
+            var newInstructions = ListPool<CodeInstruction>.Shared.Rent(instructions);
+
+            // The index offset.
+            const int offset = 0;
 
-            DoorVariant componentInParent = hitInfo.collider.gameObject.GetComponentInParent<DoorVariant>();
-            if (componentInParent == null || componentInParent.GetExactState() > 0.0 || Scp096._takenDoors.ContainsKey(__instance.Hub.gameObject))
-                return false;
+            // Search for last "lfsfld".
+            var index = newInstructions.FindLastIndex(instruction => instruction.opcode == OpCodes.Ldsfld) + offset;
 
-            var ev = new TryingNotToCryEventArgs(__instance, API.Features.Player.Get(__instance.Hub), componentInParent);
+            // Get the count to find the previous index
+            var oldCount = newInstructions.Count;
 
-            Handlers.Scp096.OnTryingNotToCry(ev);
+            // Get the return label from the instruction.
+            var returnLabel = newInstructions[index - 1].labels[0];
 
-            if (ev.IsAllowed)
+            newInstructions.InsertRange(index, new[]
             {
-                Scp096._takenDoors.Add(__instance.Hub.gameObject, componentInParent);
-                __instance.PlayerState = Scp096PlayerState.TryNotToCry;
-            }
+                new CodeInstruction(OpCodes.Ldarg_0),
+                new CodeInstruction(OpCodes.Ldarg_0),
+                new CodeInstruction(OpCodes.Ldfld, Field(typeof(PlayableScp), nameof(PlayableScp.Hub))),
+                new CodeInstruction(OpCodes.Call, Method(typeof(API.Features.Player), nameof(API.Features.Player.Get), new[] { typeof(ReferenceHub) })),
+                new CodeInstruction(OpCodes.Ldloc_1),
+                new CodeInstruction(OpCodes.Ldc_I4_1),
+                new CodeInstruction(OpCodes.Newobj, GetDeclaredConstructors(typeof(TryingNotToCryEventArgs))[0]),
+                new CodeInstruction(OpCodes.Dup),
+                new CodeInstruction(OpCodes.Call, Method(typeof(Handlers.Scp096), nameof(Handlers.Scp096.OnTryingNotToCry))),
+                new CodeInstruction(OpCodes.Call, PropertyGetter(typeof(TryingNotToCryEventArgs), nameof(TryingNotToCryEventArgs.IsAllowed))),
```

Comment:
Same as above.

Reply:
Much like above I wasn't too sure because I was looking at other patches, so I hope the C# interpretation of the IL is enough

Reply:
I was talking about using `Callvirt`.